tests/includes/handler/class-test-move.php

```diff
@@ -0,0 +1,257 @@
+		wp_delete_user( $this->user_id_2 );
+	}
+
+	/**
+	 * Test the handle_move method with a target and origin.
+	 */
+	public function test_handle_move_with_target_and_origin() {
+		$target = 'https://example.com/new-profile';
+		$origin = 'https://example.com/old-profile';
+
+		// Mock the HTTP response for the origin object.
+		$origin_object = array(
+			'type'    => 'Person',
+			'id'      => $origin,
+			'url'     => $origin,
+			'name'    => 'Old Profile',
+			'inbox'   => 'https://example.com/old-profile/inbox',
+			'movedTo' => $target,
+		);
+
+		// Mock the HTTP response for the target object.
+		$target_object = array(
+			'type'          => 'Person',
+			'id'            => $target,
+			'url'           => $target,
+			'name'          => 'New Profile',
+			'inbox'         => 'https://example.com/new-profile/inbox',
+			'also_known_as' => array(
+				$origin,
+			),
+		);
+
+		// Create a follower for the origin.
+		$origin_follower = new Follower();
+		$origin_follower->set_inbox( 'https://example.com/old-profile/inbox' );
+		$origin_follower->set_name( 'Old Profile' );
+		$origin_follower->set_type( 'Person' );
+		$origin_follower->set_id( $origin );
+		$origin_follower->set_url( $origin );
+		$id = $origin_follower->upsert();
+
+		// Add the user ID meta value.
+		add_post_meta( $id, '_activitypub_user_id', $this->user_id );
+
+		// Mock the HTTP request.
+		add_filter(
+			'pre_http_request',
+			function ( $preempt, $args, $url ) use ( $target, $target_object, $origin, $origin_object ) {
+				if ( $url === $target ) {
+					return array(
+						'body'     => wp_json_encode( $target_object ),
+						'response' => array( 'code' => 200 ),
+					);
+				}
+				if ( $url === $origin ) {
+					return array(
+						'body'     => wp_json_encode( $origin_object ),
+						'response' => array( 'code' => 200 ),
+					);
+				}
+				return $preempt;
+			},
+			10,
+			3
+		);
+
+		$activity = array(
+			'type'   => 'Move',
+			'actor'  => $origin,
+			'object' => $target,
+		);
+
+		Move::handle_move( $activity );
+
+		// Check if the origin follower was updated.
+		$updated_follower = Followers::get_follower( $this->user_id, $target );
+
+		$this->assertNotNull( $updated_follower );
+		$this->assertEquals( $target, $updated_follower->get_id() );
+		$this->assertEquals( 'https://example.com/new-profile/inbox', $updated_follower->get_inbox() );
+
+		$updated_follower->delete();
+	}
+
+	/**
+	 * Test the handle_move method with an invalid target.
+	 */
+	public function test_handle_move_with_invalid_target() {
+		$activity = array(
+			'type'   => 'Move',
+			'actor'  => 'https://example.com/old-profile',
+			'object' => 'https://example.com/invalid-profile',
+		);
+
+		// Mock the HTTP request to return an error.
+		add_filter(
+			'pre_http_request',
+			function () {
+				return new \WP_Error( 'http_request_failed', 'Invalid request' );
+			}
+		);
+
+		// Should return without error.
+		$this->assertNull( Move::handle_move( $activity ) );
```

Comment:
`handle_move()` never returns anything, so this might not be the best thing to check. Maybe verifying that it didn't change the original followers guid instead?